test/com/google/enterprise/adaptor/database/DatabaseAdaptorTest.java

```diff
@@ -561,4 +567,97 @@ public void testIncludeAllColumnsAsMetadataFalse_mcSet() throws Exception {
     assertEquals("gsa1", adaptor.metadataColumns.getMetadataName("db_col1"));
     assertEquals("gsa2", adaptor.metadataColumns.getMetadataName("col2"));
   }
+
+  private void addDBTableForTest() throws SQLException {
+    executeUpdate("create table employees("
+        + "ID" + " integer,"
+        + "FIRST_NAME" + " varchar, "
+        + "LAST_NAME" + " varchar)");
+    executeUpdate("insert into employees(ID, FIRST_NAME, LAST_NAME)"
+        + " values('1001', 'John', 'Doe')");
+  }
+
+  @Test
+  public void testGetDocIds() throws Exception {
+    addDBTableForTest();
+
+    Map<String, String> configEntries = new HashMap<String, String>();
+    configEntries.put("db.user", "sa");
+    configEntries.put("db.password", "");
+    configEntries.put("db.url",
+        "jdbc:h2:mem:test;DB_CLOSE_DELAY=-1;DEFAULT_ESCAPE=");
+    configEntries.put("db.uniqueKey", "ID:int");
+    configEntries.put("db.everyDocIdSql", "select * from employees");
+    configEntries.put("db.singleDocContentSql",
+        "select * from employees where ID = ?");
+    configEntries.put("db.singleDocContentSqlParameters", "ID");
```

Comment:
Most of the rest of these are unnecessary. I'd pick rowToText or contentColumn, which require less configuration.